salesforce/write.go

```diff
@@ -0,0 +1,130 @@
+package salesforce
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/spyzhov/ajson"
+)
+
+// Writes data to Salesforce
+func (c *Connector) Write(ctx context.Context, config common.WriteParams) (*common.WriteResult, error) {
+	var (
+		data *ajson.Node
+		err  error
+	)
+
+	location, joinErr := url.JoinPath(fmt.Sprintf("%s/sobjects", c.BaseURL), config.ObjectName)
+	if joinErr != nil {
+		return nil, joinErr
+	}
+
+	if config.ObjectId != "" {
+		location, joinErr = url.JoinPath(location, config.ObjectId)
+		if joinErr != nil {
+			return nil, joinErr
+		}
+		// Salesforce allows for PATCH method override
+		location += "?_HttpMethod=PATCH"
+	}
+	data, err = c.post(ctx, location, config.ObjectData)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return parseWriteResult(data)
+}
+
+// parseWriteResult parses the response from writing to Salesforce API. A 2xx return type is assumed.
+func parseWriteResult(data *ajson.Node) (*common.WriteResult, error) {
+
+	// in case we got a 204 and empty array => unmarshal into nil ajson node
+	if data == nil {
+		return &common.WriteResult{
+			Success: true,
+		}, nil
+	}
+
+	createdObjectId, err := getCreatedObjectId(data)
+	if err != nil {
+		return nil, err
+	}
+
+	errors, err := getErrors(data)
+	if err != nil {
+		return nil, err
+	}
+
+	success, err := getSuccess(data)
+	if err != nil {
+		return nil, err
+	}
+	return &common.WriteResult{
+		ObjectId: createdObjectId,
+		Errors:   errors,
+		Success:  success,
```

Comment:
I don't actually see where you're adding `RecordData` to the WriteResult

Reply:
That's because the Salesforce API doesn't return any written record, just the object id.

https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/dome_sobject_create.htm

Reply:
The RecordData is there for future connectors to populate